dcos/marathon.py

```diff
@@ -319,38 +319,37 @@ def add_app(self, app_resource):
 
         return response.json()
 
-    def _update(self, resource_id, payload, force=None, url_endpoint="apps"):
-        """Update an application or group.
+    def _update(self, resource_type, resource_id, resource_json, force=False):
+        """Update an application, group, or pod.
 
-        :param resource_id: the app or group id
+        :param resource_type: one of 'apps', 'groups', or 'pods'
+        :type resource_type: str
+        :param resource_id: the app, group, or pod ID
         :type resource_id: str
-        :param payload: the json payload
-        :type payload: dict
+        :param resource_json: the json payload
+        :type resource_json: {}
         :param force: whether to override running deployments
         :type force: bool
-        :param url_endpoint: resource type to update ("apps" or "groups")
-        :type url_endpoint: str
         :returns: the resulting deployment ID
         :rtype: str
         """
 
-        resource_id = util.normalize_marathon_id_path(resource_id)
-
-        if not force:
-            params = None
-        else:
-            params = {'force': 'true'}
-
-        path = 'v2/{}{}'.format(url_endpoint, resource_id)
-
-        response = self._rpc.http_req(http.put,
-                                      path,
-                                      params=params,
-                                      json=payload)
-
-        return response.json().get('deploymentId')
+        path_prefix = 'v2/{}'.format(resource_type)
+        path = self._marathon_id_path_join(path_prefix, resource_id)
+        params = self._force_params(force)
+        response = self._rpc.http_req(
+            http.put, path, params=params, json=resource_json)
+        body_json = self._parse_json(response)
 
-    def update_app(self, app_id, payload, force=None):
+        try:
+            return body_json['deploymentId']
+        except KeyError:
+            template = ('Error: missing "deploymentId" field in the following '
+                        'JSON response from\nMarathon:\n{}')
```

Comment:
Oops, I was trying to keep the output narrower than 80 chars, but now that you point it out this is probably the wrong place to do that, haha. Will remove.